packages/firebase_admob/README.md

```diff
@@ -1,3 +1,171 @@
+
+Banner and native ads in the `firebase_admob` plugin were displayed as an overlay on top of all app
+content. With `google_mobile_ads`, they can now only be displayed as a Flutter Widget. Below is an
+example of displaying a BannerAd anchored to the bottom with an offset using the `firebase_admob`
+plugin:
+
+```dart
+void main() {
+ WidgetsFlutterBinding.ensureInitialized();
+ FirebaseAdMob.instance.initialize(appId: appId);
+
+ runApp(MyApp());
+}
+
+class MyApp extends StatefulWidget {
+ @override
+ MyAppState createState() => MyAppState();
+}
+
+class MyAppState extends State<MyApp> {
+ BannerAd _bannerAd;
+
+ @override
+ void initState() {
+   super.initState();
+   _bannerAd = BannerAd(
+     adUnitId: BannerAd.testAdUnitId,
+     size: AdSize.banner,
+   );
+
+   _bannerAd
+     ..load()
+     ..show(
+       anchorOffset: 10.0,
+       anchorType: AnchorType.bottom,
+     );
+ }
+
+ @override
+ Widget build(BuildContext context) {
+   return MyAppWidget();
+ }
+}
+```
+
+When transitioning to `google_mobile_ads`, the `show` method has been removed with `AdWidget`
+replacing it. Below is an example of displaying an ad with `google_mobile_ads` that is also anchored
+to the bottom with an offset:
+
+```dart
+void main() {
+ WidgetsFlutterBinding.ensureInitialized();
+ MobileAds.instance.initialize();
+
+ runApp(MyApp());
+}
+
+class MyApp extends StatefulWidget {
+ @override
+ MyAppState createState() => MyAppState();
+}
+
+class MyAppState extends State<MyApp> {
+ BannerAd _bannerAd;
+
+ @override
+ void initState() {
+   super.initState();
+   _bannerAd = BannerAd(
+     size: AdSize.banner,
+     adUnitId: BannerAd.testAdUnitId,
+     listener: AdListener(),
+     request: AdRequest(),
+   );
+   _bannerAd.load();
+ }
+
+ @override
+ void dispose() {
+   super.dispose();
+   _bannerAd.dispose();
+ }
+
+ @override
+ Widget build(BuildContext context) {
+   return Stack(
+     children: [
+       MyAppWidget(),
+       Container(
+         padding: EdgeInsets.only(bottom: 10.0),
+         alignment: Alignment.bottomCenter,
+         child: AdWidget(ad: _bannerAd),
+       ),
+     ],
+   );
+ }
+}
+```
+
+### Interstitial and Rewarded Ads
+
+The typical pattern for displaying interstitial and rewarded Ads with `firebase_admob` would be to
+call `show` right after `load`. This is now discouraged and `show` should only be called when
+`AdListener.onAdLoaded` has been called. Below is an example usage of an interstitial ad:
+
+```dart
+class MyApp extends StatefulWidget {
+ @override
+ MyAppState createState() => MyAppState();
+}
+
+class MyAppState extends State<MyApp> {
+ InterstitialAd _interstitialAd;
+
+ @override
+ void initState() {
+   super.initState();
+   _interstitialAd = InterstitialAd(
+     adUnitId: InterstitialAd.testAdUnitId,
+     request: AdRequest(),
+     listener: AdListener(
+       onAdLoaded: (Ad ad) {
+         // Ad is now ready to show at any time.
+         // (ad as InterstitialAd).show();
```

Comment:
should it call `show`, or leave it as an option to the developer?

Reply:
It's bad practice to call `show` here. A user is supposed to preload an Ad before its needed to be shown. I'll just remove this line and leave the comment above.